pl_bolts/models/rl/ppo_model.py

```diff
@@ -0,0 +1,373 @@
+            batch_size:  batch_size when training network- can simulate number of policy updates performed per epoch
+            steps_per_epoch: how many action-state pairs to rollout for trajectory collection per epoch
+            nb_optim_iters: how many steps of gradient descent to perform on each batch
+            clip_ratio: hyperparameter for clipping in the policy objective
+        """
+        super().__init__()
+
+        if not _GYM_AVAILABLE:
+            raise ModuleNotFoundError('This Module requires gym environment which is not installed yet.')
+
+        # Hyperparameters
+        self.lr_actor = lr_actor
+        self.lr_critic = lr_critic
+        self.steps_per_epoch = steps_per_epoch
+        self.nb_optim_iters = nb_optim_iters
+        self.batch_size = batch_size
+        self.gamma = gamma
+        self.lam = lam
+        self.max_episode_len = max_episode_len
+        self.clip_ratio = clip_ratio
+        self.save_hyperparameters()
+
+        self.env = gym.make(env)
+        # value network
+        self.critic = MLP(self.env.observation_space.shape, 1)
+        # policy network (agent)
+        if isinstance(self.env.action_space, gym.spaces.box.Box):
+            act_dim = self.env.action_space.shape[0]
+            actor_mlp = MLP(self.env.observation_space.shape, act_dim)
+            self.actor = ActorContinous(actor_mlp, act_dim)
+        elif isinstance(self.env.action_space, gym.spaces.discrete.Discrete):
+            actor_mlp = MLP(self.env.observation_space.shape, self.env.action_space.n)
+            self.actor = ActorCategorical(actor_mlp)
+        else:
+            raise NotImplementedError(
+                'Env action space should be of type Box (continous) or Discrete (categorical). '
+                f'Got type: {type(self.env.action_space)}'
+            )
+
+        self.batch_states = []
+        self.batch_actions = []
+        self.batch_adv = []
+        self.batch_qvals = []
+        self.batch_logp = []
+
+        self.ep_rewards = []
+        self.ep_values = []
+        self.epoch_rewards = []
+
+        self.episode_step = 0
+        self.avg_ep_reward = 0
+        self.avg_ep_len = 0
+        self.avg_reward = 0
+
+        self.state = torch.FloatTensor(self.env.reset())
+
+    def forward(self, x: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        """
+        Passes in a state x through the network and returns the policy and a sampled action
+        Args:
+            x: environment state
+        Returns:
+            Tuple of policy and action
+        """
+        pi, action = self.actor(x)
+        value = self.critic(x)
+
+        return pi, action, value
+
+    def discount_rewards(self, rewards: List[float], discount: float) -> List[float]:
+        """Calculate the discounted rewards of all rewards in list
+        Args:
+            rewards: list of rewards/advantages
+        Returns:
+            list of discounted rewards/advantages
```

Comment:
Needs empty lines before sections to properly render docs.
```suggestion
        """Calculate the discounted rewards of all rewards in list

        Args:
            rewards: list of rewards/advantages

        Returns:
            list of discounted rewards/advantages
```